tests/e2e/helpers.go

```diff
@@ -28,21 +30,40 @@ func newHttpClient() http.Client {
 }
 
 func httpGet(url string) ([]byte, error) {
-	if !strings.Contains(url, "http") {
-		url = fmt.Sprintf("http://%s", url)
+	client := newHttpClient()
+	resp, err := client.Get(validUrl(url))
+	if err != nil {
+		return nil, err
 	}
 
+	return extractBody(resp.Body)
+}
+
+func httpPost(url string, data []byte) ([]byte, error) {
 	client := newHttpClient()
-	resp, err := client.Get(url)
+	resp, err := client.Post(validUrl(url), "application/json", bytes.NewBuffer(data))
 	if err != nil {
 		return nil, err
 	}
-	body, err := ioutil.ReadAll(resp.Body)
+
+	return extractBody(resp.Body)
+}
+
+func validUrl(url string) string {
```

Comment:
This is not really validation. Validation means throwing error and not modifying state. You can call it sanitizeUrl()

Reply:
Fixed. please approve it again.